### PR TITLE
retry repo login requests

### PR DESCRIPTION
**Pull request by amoffat**
And also refactor out a `retry_call` helper, and put that in the place where retrying was done in the push/pull method

### CODE REVIEW

File: flyingcloud/base.py [hunk 1]
```diff
@@ -761,16 +730,17 @@ def get_latest_tag(self, namespace, image_name):
         namespace.logger.info("get_latest_tag('%s') = '%s'", repo, tag)
         return tag
 
-    def docker_login(self, namespace, username, password, email=None, registry=None):
```
Comment (adamfeuer) [inline flyingcloud/base.py]: @amoffat let's leave `namespace` spelled out. We should all be using autocompletion in our editors. Spelling it out makes it easier for us to understand the code.

File: flyingcloud/base.py [hunk 2]
```diff
@@ -761,16 +730,17 @@ def get_latest_tag(self, namespace, image_name):
...
+    def docker_login(self, ns, username, password, email=None, registry=None):
         if registry:
             if username and password:
-                namespace.logger.info(
```
Comment (adamfeuer) [inline flyingcloud/base.py]: @amoffat Leave it spelled out here too

File: flyingcloud/utils/docker_util.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
```
Comment (adamfeuer) [inline flyingcloud/utils/docker_util.py:1]: @amoffat instead of this code, can we use the [retrying](https://pypi.python.org/pypi/retrying) library...?

**Review by adamfeuer — CHANGES_REQUESTED** [review]
Just some small comments - in general we try not to abbreviate in our code in order to make it more readable.

Re: the custom retrying code- was there a reason we decided not to use the retrying library?

**Comment by adamfeuer** [comment]
@amoffat Something is wrong with the travis build, the nightly python version failed. I'm looking at it.

File: flyingcloud/utils/docker_util.py (hunk 3 above)
Comment (amoffat) [reply flyingcloud/utils/docker_util.py:1]: is that like https://www.npmjs.com/package/trim ? :)  In seriousness, we're doing custom logging and catching custom exceptions.  I'm sure it is possible to do with retrying but easier to just write it, given how simple it is for our use case

**Comment by amoffat** [comment]
@adamfeuer made your change suggestions

**Comment by adamfeuer** [comment]
@amoffat LGTM 👍

**Pull request merged**